components/Payment/services/fee_calculators.rst

```diff
@@ -0,0 +1,104 @@
+Fee Calculators
+===============
+
+Fixed Fee Calculator
+--------------------
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Payment\Calculator\FixedFeeCalculator;
+   use Sylius\Component\Payment\Model\Payment;
+
+   $fixedFeeCalculator = new FixedFeeCalculator();
+
+   $payment = new Payment();
+   $payment->setAmount(500);
+
+   $configuration = array('amount' => 10);
+
+   $fixedFeeCalculator->calculate($payment, $configuration);
+   // returns 10 as it takes only the fixed value
+   // stated by configurations 'amount' key
+
+.. note::
+   This calculator implements ``FeeCalculatorInterface``
+
+Percent Fee Calculator
+----------------------
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Payment\Calculator\PercentFeeCalculator;
+   use Sylius\Component\Payment\Model\Payment;
+
+   $percentFeeCalculator = new PercentFeeCalculator();
+
+   $payment = new Payment();
+   $payment->setAmount(200);
+
+   $configuration = array('percent' => 10);
+
+   $percentFeeCalculator->calculate($payment, $configuration);
+   // returns 20 as it divides the value stated by configurations 'percent' key
+   // by 100 and then multiplies it by $payments amount property
+
+.. note::
+   This calculator implements ``FeeCalculatorInterface``
+
+Delegating Fee Calculator
+-------------------------
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Payment\Model\Payment;
+   use Sylius\Component\Payment\Model\PaymentMethod;
+   use Sylius\Component\Payment\Calculator\DelegatingFeeCalculator;
+   // also use your own service registry which implements
+   // the Sylius\Component\Registry\ServiceRegistryInterface
+
+   $serviceRegistry = // your registry, for example lets say it has the fixed
+                      // and percent calculators included
+
+   $calculator = DelegatingFeeCalculator($serviceRegistry);
+
+   $configurations = array('amount' => 14, 'percent' => 23);
+
+   $paymentMethod = new PaymentMethod();
+   $paymentMethod->setFeeCalculatorConfiguration($configurations);
+
+   $payment = new Payment();
+   $payment->setAmount(200);
+   $payment->setMethod($paymentMethod);
+
+   $calculator->calculate($payment);
+   // returns 14 as the FixedFeeCalculator is set by default in paymentMethod
+
+   $paymentMethod->setFeeCalculator(DefaultFeeCalculators::PERCENT);
+
+   $calculator->calculate($payment);
+   // now returns 46, because we changed the fee calculator
+
+.. note::
+This calculator implements ``DelegatingFeeCalculatorInterface``
+
+Default Fee Calculators
+-----------------------
+
+The following calculator types are available by default:
```

Comment:
This information can be placed on the beginning of this chapter, maybe together with some encouragement to implement own calculators?